monai/data/dataloader.py

```diff
@@ -64,6 +65,11 @@ def __init__(
         timeout: float = 0.0,
         multiprocessing_context: Optional[Callable] = None,
     ) -> None:
+        if num_workers == 0:
+            # when num_workers > 0, random states are determined by worker_init_fn
+            # this is to make the behavior consistent when num_workers == 0
+            _seed = torch.empty((), dtype=torch.int64).random_(generator=None).item()
+            set_rnd(dataset, _seed)
         super().__init__(  # type: ignore[call-overload]
```

Comment:
previously when `num_workers == 0`, the `worker_init_fn` is not used,  the loader's behaviour becomes different from those `num_workers > 0`. the modification here is to make them consistent. @Nic-Ma